### PR TITLE
fix wasm hgu

### PR DESCRIPTION
**Pull request by zjg555543**
- Targeted PR against correct branch (see [CONTRIBUTING.md](https://github.com/okx/okbchain/blob/develop/CONTRIBUTING.md#pr-targeting))

- [ ] Linked to github-issue with discussion and accepted design OR link to spec that describes this work.
- [ ] Wrote tests
- [ ] Updated relevant documentation (`docs/`)
- [ ] Added a relevant changelog entry to the `Unreleased` section in `CHANGELOG.md`
- [ ] Reviewed `Files changed` in the github PR explorer

______

For Admin Use:
- Added appropriate labels to PR (ex. wip, ready-for-review, docs)
- Reviewers Assigned
- Squashed all commits, uses message "Merge PR #XYZ: [title]" ([coding standards](https://github.com/okx/coding/blob/master/README.md#merging-a-pr))

### CODE REVIEW

**Review by yann-sjtu — APPROVED** [review]

File: x/wasm/types/tx.go [hunk 1]
```diff
@@ -78,6 +79,16 @@ func (msg MsgStoreCode) GetSigners() []sdk.AccAddress {
 	return []sdk.AccAddress{sdk.WasmToAccAddress(senderAddr)}
 }
 
+func (msg MsgStoreCode) FnSignatureInfo() (string, int, error) {
+	codeLen := len(msg.WASMByteCode)
+	var err error
+	if codeLen <= 0 {
```
Comment (giskook) [inline x/wasm/types/tx.go:85]:
```suggestion
	if codeLen == 0 {
```

File: x/wasm/types/tx.go [hunk 2]
```diff
@@ -163,6 +174,35 @@ func (msg MsgExecuteContract) GetSigners() []sdk.AccAddress {
...
 }
 
+func (msg MsgExecuteContract) FnSignatureInfo() (string, int, error) {
+	if err := msg.Msg.ValidateBasic(); err != nil {
+		return "", 0, fmt.Errorf("failed to validate msg:%v", err)
+	}
+
+	var v interface{}
+	json.Unmarshal(msg.Msg, &v)
+	data := v.(map[string]interface{})
```
Comment (giskook) [inline x/wasm/types/tx.go:184]: Maybe we should handle error both the Unmarshal and type conversion?

File: x/wasm/types/tx.go [hunk 3]
```diff
@@ -163,6 +174,35 @@ func (msg MsgExecuteContract) GetSigners() []sdk.AccAddress {
...
+		return "", 0, fmt.Errorf("failed to check msg method:%s", string(msg.Msg.Bytes()))
+	}
+
+	method := ""
+	for k, _ := range data {
+		method = k
+		break
+	}
+
+	if len(method) <= 0 {
```
Comment (giskook) [inline x/wasm/types/tx.go:195]:
```suggestion
	if len(method) == 0 {
```

File: x/wasm/types/tx.go (hunk 1 above)
Comment (zjg555543) [reply x/wasm/types/tx.go:85]: **codeLen** must eaual or larger than 0, here is compatibility.

File: x/wasm/types/tx.go (hunk 3 above)
Comment (zjg555543) [reply x/wasm/types/tx.go:195]: I's can be compatibility and robustness.

**Review by giskook — APPROVED** [review]

**Pull request merged**